crates/matrix-sdk-crypto/src/types/events/utd_cause.rs

```diff
@@ -163,6 +190,46 @@ impl UtdCause {
             _ => UtdCause::Unknown,
         }
     }
+
+    /**
+     * Below is the flow chart we follow for deciding whether historical
+     * UTDs are expected. This function starts at position `B`.
+     *
+     * ```mermaid
+     * flowchart TD
+     * A{Is the message newer than the device?}
+     * A -- No --> B
+     * A -- Yes --> X
+     * B{Is there a backup on the server?}
+     * B -- No --> Y
+     * B -- Yes --> C
+     * C{Is backup working on this device?}
+     * C -- No --> D
+     * C -- Yes --> X
+     * D{Is this device verified?}
+     * D -- No --> Z
+     * D -- Yes --> X
+     *
+     * X([Normal UTD error])
+     * Y([History is not available on this device])
+     * Z([You need to verify this device])
+     * ```
+     */
+    fn determine_historical(crypto_context_info: CryptoContextInfo) -> UtdCause {
+        if crypto_context_info.backup_exists_on_server {
+            if crypto_context_info.is_backup_configured
+                || crypto_context_info.this_device_is_verified
+            {
+                // Either backup download is working but we still don't have the key, or backup download
+                // *isn't* working despite being verified. In either case, we shrug and give an `Unknown` cause.
```

Comment:
As discussed, I think this is inconsistent with web were any device historical UTD is reported using `HistoricalMessage`. Here EX will report those as `OlmKeysNotSentError` https://github.com/element-hq/element-x-ios/blob/9856e3e5b45d3896eba7b8cd84f4cedfffc92d3f/ElementX/Sources/FlowCoordinators/UserSessionFlowCoordinator.swift#L387

Reply:
btw I was working from https://github.com/element-hq/element-meta/issues/2313#issuecomment-2507713793 which was my modification of the flowchart above that Rich made. We should figure this stuff out and write it down somewhere sensible.

Reply:
> As discussed, I think this is inconsistent with web were any device historical UTD is reported using HistoricalMessage.

I must say, that doesn't match my understanding. I thought HistoricalMessage was reserved for the *expected* UTD case of a historical message with no backup on the server.

Reply:
Yes @BillCarsonFr this is why I think what I implemented matches Web:

From [DecryptionFailureBody.getErrorMessage](https://github.com/element-hq/element-web/blob/d68c5a26af1aa8e17300eb3e0d3f76a804f89c3d/src/components/views/messages/DecryptionFailureBody.tsx#L19)

![image](https://github.com/user-attachments/assets/01f881e0-1952-4a86-a96c-8a8a65d1a809)

It handles 3 historical cases: no backup exists, backup fails and we are unverified, and everything else - "unable to decrypt".

Reply:
We agreed in real life that this change is an improvement even though it's not the final solution, so we are OK to land it.